### PR TITLE
Remove support for go.tls.config

### PR DESCRIPTION
**Pull request by Strainy**
This change removes the TLS callbacks we introduced in our [forked version of librkafka](https://github.com/zendesk/librdkafka/tree/zendesk_release).

This allows us to maintain compatibility with the official version of librdkafka, facilitating a hopefully simpler maintenance process for zendesk_kafka_go V2.

For context to why this change is necessary, please refer to the following Slack thread:
https://zendesk.slack.com/archives/C12KJMEFM/p1657173565878259

### CODE REVIEW

File: kafka/librdkafka_vendor/README.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+> $ git remote add upstream https://github.com/confluentinc/confluent-kafka-go
+> ...
+> $ git fetch --all
+> ... 
+> $ git checkout v1.9.0 -- kafka/librdkafka_vendor
+> ...
+> # still need to regenerate error codes
+> $ make -f mk/Makefile generr
+> ...
+> ```
```
Comment (Strainy) [inline kafka/librdkafka_vendor/README.md]: Just noting a modified process that we can use here for checking out the updated vendored librdkafka upstream.

File: kafka/build_darwin.go [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-
 // #cgo CFLAGS: -DUSE_VENDORED_LIBRDKAFKA -DLIBRDKAFKA_STATICLIB
-// #cgo LDFLAGS: ${SRCDIR}/librdkafka_vendor/librdkafka_darwin.a  -lz -lm -lsasl2 -ldl -lpthread
+// #cgo LDFLAGS: ${SRCDIR}/librdkafka_vendor/librdkafka_darwin.a  -lm -lsasl2 -ldl -lpthread -framework CoreFoundation -framework SystemConfiguration
 import "C"
 
 // LibrdkafkaLinkInfo explains how librdkafka was linked to the Go client
-const LibrdkafkaLinkInfo = "static darwin from librdkafka-static-bundle-v1.7.0.tgz"
-
- */
```
Comment (Strainy) [inline kafka/build_darwin.go]: Just having a play with how we might bring back the static linking here. I think we'll want to do this is a separate PR though.

**Review by yunhan0 — APPROVED** [review]
👍 LGTM

**Review by cweerasooriya — APPROVED** [review]

**Review by Jesse-Cameron — APPROVED** [review]
Looks good! Thanks! :)
